lib/lightning_web/live/workflow_live/components.ex

```diff
@@ -800,4 +800,61 @@ defmodule LightningWeb.WorkflowLive.Components do
     </.modal>
     """
   end
+
+  def workflow_metrics(assigns) do
+    ~H"""
+    <div class="grid gap-4 md:grid-cols-2 lg:grid-cols-4">
+      <%= for {key,value} <- assigns.metrics do %>
+        <%= case key do %>
+          <% :total_workorders -> %>
+            <.metric_card title="Work Orders" body={value} />
+          <% :total_runs -> %>
+            <.metric_card
+              title="Runs"
+              body={Map.get(value, :completed_runs)}
+              optional_text={
+                Map.get(value, :pending_running_runs)
+                |> (to_string |> Kernel.<>(" pending"))
+              }
+            />
+          <% :successful_runs -> %>
+            <.metric_card
+              title="Successful Runs"
+              body={Map.get(value, :successful_runs)}
+              optional_text={
+                Map.get(value, :success_percentage) |> (to_string |> Kernel.<>("%"))
+              }
+            />
+          <% :failed_workorders -> %>
+            <.metric_card
+              title="Work Orders in failed state"
+              body={Map.get(value, :failed_workorders)}
+              optional_text={
+                Map.get(value, :failure_percentage) |> (to_string |> Kernel.<>("%"))
+              }
+            />
+          <% _-> %>
+        <% end %>
+      <% end %>
```

Comment:
Why is this in a loop? We have a known number of cards, and looking at the case statement, every match requires different attributes.

Also I think putting the optional text in an expression might invalidate LiveViews diffing. I'd suggest changing `optional_text` to `suffix`. _or_ better yet, use a named slot for both the main and suffix section. So instead of passing in the body as text use a slot.

```heex
<.metric_card ...>
   <:value>{@metrics.failed_wordorders}</:value>
   <:suffix>({@metrics.failure_percentage}%)</:suffix>
</.metric_card>
```

Also please don't use the `assigns` variable (or any other local variable) inside a template string. See https://hexdocs.pm/phoenix_live_view/assigns-eex.html#pitfalls

Reply:
Resolved. Thank you